pkg/guestagent/events/containerd_linux.go

```diff
@@ -0,0 +1,365 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/errdefs"
+	containerdNamespace "github.com/containerd/containerd/namespaces"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/lima-vm/lima/v2/pkg/guestagent/api"
+)
+
+const (
+	stateKey             = "nerdctl/state-dir"
+	portsKey             = "nerdctl/ports"
+	namespaceKey         = "nerdctl/namespace"
+	defaultSocketTimeout = 5 * time.Second
+)
+
+type ipPortMap map[string][]*api.IPPort
+
+type ContainerdEventMonitor struct {
+	socketPaths []string
+}
+
+func NewContainerdEventMonitor(socketPaths []string) *ContainerdEventMonitor {
+	return &ContainerdEventMonitor{
+		socketPaths: socketPaths,
+	}
+}
+
+func (c *ContainerdEventMonitor) MonitorPorts(ctx context.Context, ch chan *api.Event, errCh chan error) {
+	const defaultRetryDelay = 2
+	retryDelay := defaultRetryDelay
```

Comment:
Shouldn't the initial delay be 0? We don't want to wait the first time we enter the loop.

```suggestion
	retryDelay := 0
```